src/Core/Grid/Position/PositionDataHandler.php

```diff
@@ -0,0 +1,151 @@
+<?php
+/**
+ * 2007-2018 PrestaShop.
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2018 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Core\Grid\Position;
+
+use PrestaShop\PrestaShop\Core\Grid\Position\Exception\PositionDataException;
+
+/**
+ * Class PositionDataHandler is a basic implementation of the PositionDataHandlerInterface,
+ * it transforms the provided array data into a PositionUpdate object.
+ * Of course, you need to respect a data format for it to work but if you don't want to use
+ * this format you are free to implement your own interface.
+ */
+class PositionDataHandler implements PositionDataHandlerInterface
+{
+    /**
```

Comment:
As I prefer more docs than "no docs", I'm not really easy with this huge block of doc.

Also - I don't think we should do it now - if you need a very specific format, why not use a specific object instead of an array?

Something like TransationalObjectToBeNamed, and you can have a specific constructor that takes an array and the constructor is in charge of validating the data.

```php
try {
    $validData = TransationalObjectToBeNamed::createFromArray($data);
    $positionUpdateFactory->getPositionUpdate($validData);
}catch(PositionDataException) {
    // not sure we should catch it but still...
}
```

This will extract the validation of your input data outside of this class, which is also more solid.

Why I'm telling this is ok to not do it "right now"? 

Because your validation process is not part of the public API and we can improve the interface later :+1:

Reply:
I'm not sure if I understand all your suggestions, but I think it is overkill to add another layer, the purpose of this DataHandler is to transfer raw array data into a PositionUpdate
so I think it's weird to add a component that transforms your data.. so that you can transform the data :P
Also this class was seen as a helper to show how a PositionDataHandlerInterface could be implemented so it is natural that it has a specific format

But maybe we could try and generalize the behaviour of this class and allow to set the different fields needed to parse the data

```
public function __construct($positionsField, $rowIdField, $oldPositionField, $newPositionField)
```

which would be injected in the services config, wdyt?

However I really like your naming 'PositionUpdatefactory' I am gonna change it